src/Middleware/HttpLogging/src/HttpLoggingServicesExtensions.cs

```diff
@@ -31,5 +31,26 @@ public static IServiceCollection AddHttpLogging(this IServiceCollection services
             services.Configure(configureOptions);
             return services;
         }
+
+        /// <summary>
+        /// Adds W3C Logging services.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> for adding services.</param>
+        /// <param name="configureOptions">A delegate to configure the <see cref="W3CLoggerOptions"/>.</param>
+        /// <returns></returns>
+        public static IServiceCollection AddW3CLogging(this IServiceCollection services, Action<W3CLoggerOptions> configureOptions)
```

Comment:
Just throwing an idea. There could be an `IStartupfilter` which adds the middleware at the first of the pipeline so there's no need to call `app.UseW3cLogging`.

Reply:
Why would we do that for this middleware?